---
title: "Figure 2: the skin is a reservoir for antibiotic resistance"
author: "Diana Proctor"
date: "2023-04-06"
output: html_document
---
***

last updated: `r format(Sys.Date(), format="%B %d %Y")`

*draft: submission to Nature, August 2023*

*Manuscript Title*: Fungal Candida auris and bacterial ESKAPE pathogens:  Human skin as a reservoir for transmissible microbes and antibiotic resistance in nursing homes

*Authors*: Diana M. Proctor 1, Sarah E Sansom2, Clay Deming1, Sean Conlan1, Ryan A Blaustein1,3, Thomas K. Atkins1,4, NISC Comparative Sequencing Program5, Thelma Dangana2, Christine Fukuda2, Lahari Thotapalli2, Heidi H. Kong6, Michael Y. Lin2, Mary K. Hayden2+ and Julia A. Segre 1+

*Affiliations*:

1Microbial Genomics Section, Translational and Functional Genomics Branch, National Human Genome Research Institute, National Institutes of Health, Bethesda, MD 20892, USA

2 Department of Internal Medicine, Division of Infectious Diseases, Rush University Medical Center, Chicago, IL 60612, USA.

3 Current Address: Department of Nutrition and Food Science, University of Maryland, College Park, MD 20742, USA

4 Current Address: Department of Quantitative and Computational Biology, Princeton University, Princeton, NJ

5 NIH Intramural Sequencing Center, National Human Genome Research Institute, National Institutes of Health, Bethesda, MD 20892, USA.

6 Dermatology Branch, National Institute of Arthritis and Musculoskeletal and Skin Diseases, National Institutes of Health, Bethesda, MD 20892, USA


+Contributed equally

---
***
The input for this is the following files, generated using the mags.sh script:

1. mapping_file_2023-07-23.csv, the metadata for the shotgun data
2.gtdbtk.arcbaceuk_nature.summary_withabx.csv, which is a blend of the GATK taxonomic output (generated from mags.sh) and the mash/mummer output (generated from mags.sh); I manually concatenated these two taxonomy files 
3. ESKAPE_phyloseq_50percent.rds, a phyloseq object describing abundance of species per sample where at least 50% of genome is covered, generated in the script for figure 1 with the breadth of coverage filter set to 50%
4. srst2_take3.txt, output of srst2
5. Cauris_MDRO history.csv, the history of antibiotic resistance mechanisms 
6. ESKAPE_phyloseq_nocoveragethreshold.rds, the phyloseq object where abundance is defined by total unique reads mapping irrespective of breadth of genome coverage, generated in the script for figure 1 with the breadth of coverage filter set to 0%
7. no_MAGs.csv, the file that indicates the subjects for whom we didn't get MAGs for the species defined in bacteria_of_interest

This script will generate the following plots:

1. Figure 2
2. Supplementary Figure 2
3. Supplementary Figure 3

First let's load needed R packages.
```{r, echo=FALSE}
#set global knitting options
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE, error = FALSE, fig.width = 12, fig.height = 7)
```




```{r, echo=FALSE}
# load package method from from Dan Sprokett

# set seed
set.seed(78979)
#define packages
packages <- c("knitr", 
              "vcfR", 
              "ANCOMBC",
              "ggplot2", 
              "tidyverse", 
              "colorspace", 
              "kableExtra", 
              "gridExtra", 
              "scales",
              "refGenome",
              "RColorBrewer",
              "ape",
              "ggtree",
              "ape",
              "Biostrings",
              "vcfR",
              "adegenet",
              "poppr",
              "pegas",
              "dendextend",
              "cowplot",
              "ggpubr",
              "ComplexHeatmap",
              "phyloseq",
              "reshape2",
              "viridis",
              "ggpmisc")
# install packages from bioconductor
BiocManager::install(setdiff(packages,installed.packages()), update=FALSE)
n <- length(packages) - sum(sapply(packages,require,character.only=TRUE))

# print if packages loaded properly
if(n == 0){
  print("All necessary R packages loaded properly")
} else {
  print(paste0(n, " R packages did not load properly"))
}
```

define the taxa of interest for the figure
```{r}
#define the taxa of interest
taxa_of_interest = c("Candida auris" ,
      "s__Escherichia coli",
      "s__Enterococcus faecalis",
     "s__Klebsiella pneumoniae" ,
     "s__Acinetobacter baumannii",
      "s__Pseudomonas aeruginosa",
      "s__Proteus mirabilis",
      "s__Providencia stuartii",
     "s__Morganella morganii",
     "s__Staphylococcus aureus")

bacteria_of_interest = c("s__Escherichia coli",
      "s__Enterococcus faecalis",
     "s__Klebsiella pneumoniae" ,
     "s__Acinetobacter baumannii",
      "s__Pseudomonas aeruginosa",
      "s__Proteus mirabilis",
      "s__Providencia stuartii",
     "s__Morganella morganii",
     "s__Staphylococcus aureus")
```

### Let's read in the data that we will use
```{r}
#read in the metagenomic data mapping file
map = read.csv("~/Desktop/NATURE/mapping_file_2023-07-23.csv")
mysubjects = unique(map$Unique_ptid)
#now read in the gtdbk classifications, so we can identify which species we find resistance genes in using abricate
gtdbk = read.csv("~/Desktop/NATURE/taxonomy/gtdbtk.arcbaceuk_nature.summary_withabx.csv") %>%
  dplyr::select(., c("user_genome", "msa_percent", "phylum", "genus", "species"))

#add a variable to gtdbk that allows us to merge it with the metagenomic data file
gtdbk$sample = stringr::str_sub(gtdbk$user_genome, 1, 7)
map$sample = map$library_id

#merge all these data frames with each other
df = plyr::join(map, gtdbk) %>%
  subset(., species %in% taxa_of_interest)

```




now let's determine which samples/subjects were associated with MAGs of each of the select species.

```{r, fig.height=5, fig.width=5}
#let's cross tabulate the taxonomy and sample information for the species of interest
crossTab = data.frame(table(df$species,df$Sample)) %>%
    subset(., Var1 %in% taxa_of_interest)

#let's revalue the Malassezias as Malassezia spp.
crossTab$group = plyr::revalue(crossTab$Var1, c("Malassezia arunalokei" ="Malassezia spp.", 
     "Malassezia globosa"="Malassezia spp.", 
     "Malassezia restricta"="Malassezia spp.", 
     "Malassezia furfur"="Malassezia spp.",
     "uncultured Malassezia"="Malassezia spp."))




#clean up the variables by removing s__ from the species names
crossTab$group = stringr::str_remove_all(crossTab$group , "s__")

     
#regenerate a mapping file from the crosstabulation and subset on the sites of interest
newmap = reshape2::colsplit(crossTab$Var2, ";", c("subject", "site", "survey"))
crossTab = data.frame(crossTab, newmap) %>%
  subset(., !(site %in% c("Site_Bu/To" ,"Site_Ne",    "Site_Ctrl", "Site_Tc")))


#revalue the body sites so that the names are consistent with the other figures
crossTab$Body_Site = plyr::revalue(crossTab$site,
                              c("Site_An" = "Peri-anal",
                                "Site_Ic"="Inguinal crease", 
                                "Site_N"="Nares",
                                "Site_Fg"="Fingertips",
                                "Site_Tw"= "Toe web",
                                "Site_Ea"="Ear Canal"))

#we need to correct for the fact that we have spades assemblies AND megahit assemblies for a select number of samples
#replace any count equal to two with a 1
crossTab$Freq = ifelse(crossTab$Freq==2, 1, crossTab$Freq)


#let's now look at how many times we assembled each of the select species out of samples for each subject. We will normalize by the number of samples for each subject.
prevDF = doBy::summary_by(crossTab, Freq~Var1+subject, FUN=c(sum, length)) 
prevDF$Unique_ptid = as.numeric(stringr::str_sub(prevDF$subject, 9, 10))

#now make a bar of prevalence
burden = prevDF
burden$Prev = ifelse(burden$Freq.sum==0, 0, 1)
burden2 = doBy::summary_by(burden, Prev~subject, FUN=c(sum, length)) 

ordering = c(1:5, 11:18, 23:38, 40:54)

burden2$Unique_ptid = stringr::str_remove_all(burden2$subject, "Subject_")
burden2$Unique_ptid <- factor(burden2$Unique_ptid, levels = ordering)

# Plot
Fig2Ap1a = ggplot(burden2, aes(x=as.factor(Unique_ptid), y=Prev.sum)) +
  geom_segment( aes(x=Unique_ptid, xend=Unique_ptid, y=0, yend=Prev.sum), color="grey") +
  geom_point( color="#9ECAE1", size=4) +
  theme_light() +
  theme(
    panel.grid.major.x = element_blank(),
    panel.border = element_blank(),
    axis.ticks.x = element_blank()
  ) +
  xlab("Subject") +
  ylab("Number of Species Detected")
Fig2Ap1a

#what is the median number of eskape pathogens carried
summary(burden2$Prev.sum)
```





### Fig2A
 Recovery of MAGs for species (y-axis) for each subject (x-axis) for skin sites, excluding peri-anal samples. Asterisk indicate subjects for whom MAGs were recovered for each species. Colors map to subjects and species where genomes were at least 50% covered (high) vs. samples and species where coverage was less than 50%. 

figure is missing subject 6, 9, and 19, 22, 39 
```{r}
#let's read in the results of the read mapping (unique reads) obtained by bwa-mem
#in order for an abundance value to be present in this table, that species must have had at least 50% of its genome covered in that sample
phy = readRDS("~/Desktop/NATURE/Figure1/ESKAPE_phyloseq_50percent.rds") %>%
  subset_taxa(., species %in% bacteria_of_interest) %>%
  merge_samples(., "Unique_ptid")
  
#make a data frame of the phyloseq object and clean it up
HighCov = data.frame(t(otu_table(phy)) , tax_table(phy))
HighCovDF = melt(HighCov, colnames(tax_table(phy))) %>%
  dplyr::select(., c("species", "variable", "value"))
colnames(HighCovDF) = c("Var1", "Unique_ptid", "Abundance")
HighCovDF$Unique_ptid = stringr::str_remove_all(HighCovDF$Unique_ptid , "X")
HighCovDF$subject = paste0("Subject_", HighCovDF$Unique_ptid )
HighCovDF$HighCov = ifelse(HighCovDF$Abundance > 0, 1, 0)
HighCovDF = dplyr::select(HighCovDF, "Unique_ptid", "Var1", "HighCov")
HighCovDF$HighCov =  ifelse(HighCovDF$HighCov > 0, "Presence", "Absence")
HighCovDF$Unique_ptid = as.numeric(as.character(HighCovDF$Unique_ptid))



#merge the coverage information with the prevalence information generated in the previous chunk
PA = prevDF
PA$prev =  ifelse(PA$Freq.sum > 0, "Yes", "No")
PA$prev = as.factor(PA$prev)
foo = colsplit(PA$subject, "_",c("Junk", "Unique_ptid"))
PA_DF = data.frame(PA, foo) %>%
  dplyr::select(., c("Unique_ptid", "Var1", "prev"))

#make a data frame for the subjects for which we have coverage but no MAGs
noMAGs = read.csv("~/Desktop/NATURE/ABX_SAMPLES/Manuscript/no_MAGs.csv")[,1:3]

PA_DF = data.frame(rbind(noMAGs, PA_DF))
PA_DF$Unique_ptid = as.numeric(as.character(PA_DF$Unique_ptid))

#merge coverage and presence/absence
FinalDF = dplyr::full_join(PA_DF, HighCovDF) 
FinalDF$Var1 = stringr::str_remove_all(FinalDF$Var1, "s__")
FinalDF$MAG = FinalDF$prev
#FinalDF = subset(FinalDF, Var1 !="0")
ordering = c(1:55)
FinalDF$Unique_ptid <- factor(FinalDF$Unique_ptid, levels = ordering)
FinalDF = subset(FinalDF, Var1 != "Candida auris")
#create an annotation data frame for whether or not we recovered a mag for a sample
annotation = subset(FinalDF, MAG=="Yes")
annotation$annotation = "*"
newcolours=c("Presence"= "midnightblue", "Absence"= "lightskyblue4" ) 

#make figure 2A
Fig2A = ggplot(FinalDF, aes(as.factor(Unique_ptid), Var1))+ 
  geom_tile(aes(fill=HighCov, color=HighCov)) + 
  geom_text(data=annotation, aes( x=as.factor(Unique_ptid), y=Var1, label=annotation),                 , 
           color="white", fontface="bold" ) + 
  scale_fill_manual(values=newcolours, name = "Legend") +
  scale_color_manual(values=newcolours, name = "Legend") +
  theme_classic() + xlab("Subject") + 
  theme(axis.text.y = element_text(face = "italic")) + ylab("") + 
  theme(legend.position="bottom") +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) 
Fig2A
```

### Supplementary Figure 2
Supplementary Figure 2: Recovery of MAGs for species (y-axis) from each subject (x-axis) for skin sites excluding peri-anal samples. Asterisks indicate subjects for whom MAGs were recovered for each species. Colors map to subjects and species where genomes were at least 50% covered (high) vs. samples and species where coverage was less than 50%. Coverage was determined by bwa-mem. 
```{r}
#let's read in the results of the read mapping (unique reads) obtained by bwa-mem
#in order for an abundance value to be present in this table, that species must have had at least 50% of its genome covered in that sample but this time subset on body site and exclude peri-anus
phy = readRDS("~/Desktop/NATURE/Figure1/ESKAPE_phyloseq_50percent.rds") %>%
  subset_taxa(., species %in% taxa_of_interest) %>%
  subset_samples(., site_specific != "An") %>%
  microViz::tax_transform(., trans = "binary", rank = "species") %>%
  merge_samples(., "Unique_ptid", fun=sum)
  
#make a data frame of the phyloseq object and clean it up
HighCov = data.frame(t(otu_table(phy)) , tax_table(phy))
HighCovDF = melt(HighCov, colnames(tax_table(phy))) %>%
  dplyr::select(., c("species", "variable", "value"))
colnames(HighCovDF) = c("Var1", "Unique_ptid", "Abundance")
HighCovDF$Unique_ptid = stringr::str_remove_all(HighCovDF$Unique_ptid , "X")
HighCovDF$subject = paste0("Subject_", HighCovDF$Unique_ptid )
HighCovDF$HighCov = ifelse(HighCovDF$Abundance > 0, 1, 0)
HighCovDF = dplyr::select(HighCovDF, "Unique_ptid", "Var1", "HighCov")
HighCovDF$HighCov =  ifelse(HighCovDF$HighCov > 0, "Presence", "Absence")
HighCovDF$Unique_ptid = as.numeric(as.character(HighCovDF$Unique_ptid))


### now let's look at the prevalence of MAGs in the absence of the peri-anal samples
#let's cross tabulate the taxonomy and sample information for the species of interest
crossTab = data.frame(table(df$species,df$Sample)) %>%
    subset(., Var1 %in% bacteria_of_interest)


#clean up the variables by removing s__ from the species names
crossTab$group = stringr::str_remove_all(crossTab$Var1 , "s__")

     
#regenerate a mapping file from the crosstabulation and subset on the sites of interest
newmap = reshape2::colsplit(crossTab$Var2, ";", c("subject", "site", "survey"))
crossTab = data.frame(crossTab, newmap) %>%
  subset(., !(site %in% c("Site_Bu/To" ,"Site_Ne",    "Site_Ctrl", "Site_Tc")))

crossTab = subset(crossTab, site !=  "Site_An")
#revalue the body sites so that the names are consistent with the other figures
crossTab$Body_Site = plyr::revalue(crossTab$site,
                              c("Site_Ic"="Inguinal crease", 
                                "Site_N"="Nares",
                                "Site_Fg"="Fingertips",
                                "Site_Tw"= "Toe web",
                                "Site_Ea"="Ear Canal"))

#we need to correct for the fact that we have spades assemblies AND megahit assemblies for a select number of samples
#replace any count equal to two with a 1
crossTab$Freq = ifelse(crossTab$Freq==2, 1, crossTab$Freq)


#let's now look at how many times we assembled each of the select species out of samples for each subject. We will normalize by the number of samples for each subject.
prevDF = doBy::summary_by(crossTab, Freq~Var1+subject, FUN=c(sum, length)) 
prevDF$Unique_ptid = as.numeric(stringr::str_sub(prevDF$subject, 9, 10))

#merge the coverage information with the prevalence information generated in the previous chunk
PA = prevDF
PA$prev =  ifelse(PA$Freq.sum > 0, "Yes", "No")
PA$prev = as.factor(PA$prev)
foo = colsplit(PA$subject, "_",c("Junk", "Unique_ptid"))
PA_DF = data.frame(PA, foo) %>%
  dplyr::select(., c("Unique_ptid", "Var1", "prev"))

#make a data frame for the subjects for which we have coverage but no MAGs
noMAGs = read.csv("~/Desktop/NATURE/ABX_SAMPLES/Manuscript/no_MAGs.csv")[,1:3]

PA_DF = data.frame(rbind(noMAGs, PA_DF))
PA_DF$Unique_ptid = as.numeric(as.character(PA_DF$Unique_ptid))

#merge coverage and presence/absence
FinalDF = dplyr::full_join(PA_DF, HighCovDF) 
FinalDF$Var1 = stringr::str_remove_all(FinalDF$Var1, "s__")
FinalDF$MAG = FinalDF$prev
#FinalDF = subset(FinalDF, Var1 !="0")
ordering = c(1:55)
FinalDF$Unique_ptid <- factor(FinalDF$Unique_ptid, levels = ordering)
FinalDF = subset(FinalDF, Var1 != "Candida auris")
#create an annotation data frame for whether or not we recovered a mag for a sample
annotation = subset(FinalDF, MAG=="Yes")
annotation$annotation = "*"
newcolours=c("Presence"= "midnightblue", "Absence"= "lightskyblue4" ) 

#make SFig2
SFig2 = ggplot(FinalDF, aes(as.factor(Unique_ptid), Var1))+ 
  geom_tile(aes(fill=HighCov, color=HighCov)) + 
  geom_text(data=annotation, aes( x=as.factor(Unique_ptid), y=Var1, label=annotation),               
           color="white", fontface="bold" ) +
  scale_fill_manual(values=newcolours, name = "Legend") +
  scale_color_manual(values=newcolours, name = "Legend") +
  theme_classic() + xlab("Subject") + 
  theme(axis.text.y = element_text(face = "italic")) + ylab("") + 
  theme(legend.position="bottom") +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) 
SFig2
ggsave(SFig2, file="~/Desktop/NATURE/ABX_SAMPLES/Manuscript/SupplementaryFigure2.png", height=4, width=12, device="png")
```



### Figure 2B
Detection of 3 carbapenemase producing genes (VIM, KPC, NDM) identified in clinical microbiology findings or surveillance cultures marked as ‘surveillance’ on x-axis. Detection of antibiotic resistance genes by read mapping in shotgun metagenomic data is marked as positive in ‘sequencing’. Each panel represents a unique subject.

okay, now let's plot the antibiotic resistance data, derived from srst2, a read-based approach.

```{r, fig.height=8, fig.width=6}
#read srst2 in
srst = read.csv("~/Desktop/NATURE/ABX_SAMPLES/srst2/srst2_take3.csv")
colnames(srst)[1] = "library_id"

#merge srst2 with the sample metadata
srstDF = merge(srst, map)
#write.csv(srstDF,"~/Desktop/NATURE/srstDF_annotated_withabx.csv")

#define the genes we're interested in seeing
Blas = c("CfxA_Bla", "PC1_Bla", "TEM_Bla", "ACI_Bla" ,"CepA_Bla" ,         "OXA_Bla" ,          "PDC_Bla",  "KPC_Bla",
         "SHV-OKP-LEN_Bla" ,"DHA_Bla","VEB_Bla" ,"ADC_Bla", "CTX-M_Bla",  "CKO_Bla" ,
         "AmpC1_Bla" ,  "BIL_Bla" ,"CMY_Bla"   ,  "SRT_Bla"  , "CblA_Bla" ,  
         "ACT_Bla" , "VIM_Bla" ,  "NDM_Bla"  )

#define the genes we're interested in seeing
Blas2 = c( "KPC_Bla", "VIM_Bla" ,  "NDM_Bla"  )

#subset the srst2 results on just the carbepenamse genes since we have surveillance data for them
carbs = subset(srstDF, gene %in% Blas2) 
carbs$gene = stringr::str_sub(carbs$gene, 1, 3)
carbs = subset(carbs, gene %in% c("KPC", "NDM", "VIM"))
carbs = dplyr::select(carbs, c("Unique_ptid", "Survey_Period", "site_specific", "gene"))
#create a variable in the sequencing results to denote positivity and identification type
carbs$is.positive = "+"
carbs$Type = "Sequencing"
SeqPositiveSubjects = unique(carbs$Unique_ptid)


#read in the surveillance/clinical microbiology findings
PCRpositives = read.csv("~/Desktop/NATURE/new_clinical_data_2023/Cauris_MDRO history.csv") 
#subset on the findings with a mechanism of action and annotate
PCRpositives = dplyr::select(PCRpositives, c("Unique_ptid","Mechanism", "Sample_type"))%>%
  subset(., Mechanism !="Unknown")
colnames(PCRpositives)[3] = "site_specific"
PCRpositives$Survey_Period = ""
colnames(PCRpositives)[2] = "gene"
PCRpositives$is.positive = "+"
PCRpositives$Type = "Surveillance"

#make a data frame combining sequencing results and the surveillance findings and subset on the patients we analyzed by shotgun
mydf = data.frame(rbind(PCRpositives, carbs)) %>%
  subset(., Unique_ptid %in% SeqPositiveSubjects)

#let's add values for Surveillance for subjects for whom we get sequencing results
s46 = c(46, "KPC", "", "", "", "Surveillance")
s53 = c(53, "KPC", "", "", "", "Surveillance")
s54 = c(54, "KPC", "", "", "", "Surveillance")

newdf = data.frame(rbind(mydf, s46, s53, s54))

ordering = c("Surveillance", "Sequencing"  )
newdf$Type <- factor(newdf$Type, levels = ordering)

#write.csv(newdf, file="~/Desktop/NATURE/ABX_SAMPLES/surveillance_sequencing.csv")
#make a plot for antibiotic genes
Fig2B = ggplot(newdf, aes(Type, gene, label=is.positive)) + geom_text() +
  facet_wrap(~Unique_ptid, scales="free_x", strip.position="left", ncol=1) +
   ylab("")+
   xlab("") + theme_classic() +
  theme(strip.background = element_blank(),
    strip.placement = "outside",
    strip.text.y = element_text(size=8, face="bold", angle=0),
     strip.text.x = element_text(size=8, face="bold", angle=0)) + 
  theme(axis.text = element_text(size = 8)) 

Fig2B
```



```{r}
mydf = data.frame(rbind(PCRpositives, carbs)) %>%
  subset(., Unique_ptid %in% mysubjects)


#let's add values for Surveillance for subjects for whom we get sequencing results
s46 = c(46, "KPC", "", "", "", "Surveillance")
s53 = c(53, "KPC", "", "", "", "Surveillance")
s54 = c(54, "KPC", "", "", "", "Surveillance")

newdf = data.frame(rbind(mydf, s46, s53, s54))

ordering = c("Surveillance", "Sequencing"  )
newdf$Type <- factor(newdf$Type, levels = ordering)
newdf = subset(newdf, gene != "NDM-1")
#write.csv(newdf, file="~/Desktop/NATURE/ABX_SAMPLES/surveillance_sequencing.csv")
#make a plot for antibiotic genes
Fig2B_All = ggplot(newdf, aes(Type, gene, label=is.positive)) + geom_text() +
  facet_wrap(~Unique_ptid, scales="free_x", strip.position="left", ncol=2) +
   ylab("")+
   xlab("") + theme_classic() +
  theme(strip.background = element_blank(),
    strip.placement = "outside",
    strip.text.y = element_text(size=8, face="bold", angle=0),
     strip.text.x = element_text(size=8, face="bold", angle=0)) + 
  theme(axis.text = element_text(size = 8)) 

```

### SFig3: how persistent are these genes over time

Supplementary Figure 3: Detection of bla-KPC at multiple time points across subjects and skin sites. Each panel represents a subject. Body sites are displayed on the y-axis and time point is displayed on the x-axis. Orange boxes are displayed where a sample was sequenced for that subject, site, and time point. Crosses are displayed in cases where bla-KPC was detected by SRST2. Persistent detection of bla-KPC is observed in subject 14, 23, 31, 53, and 54 with potential persistence observed in subjects 27 and 46. Persistence is defined as the detection of a gene at consecutive monthly surveys. Tw, toe webs; N, nares; Ic, inguinal crease; Fg, Fingertips/palm; An, peri-anus.
```{r, fig.height=4}
mymap = subset(map, Unique_ptid %in% carbs$Unique_ptid)
mymap = subset(mymap, Unique_ptid !=15)
newdf = subset(newdf, Type=="Sequencing")

#let's subset on KPC and plot
kpc = subset(newdf, gene=="KPC")
kpc$Survey_Period = as.numeric(as.character(kpc$Survey_Period))

#subset on the people who were sequencing positive
kpc = subset(kpc, Unique_ptid %in% carbs$Unique_ptid) 
kpc = subset(kpc, Unique_ptid != 15)
SFig3 = ggplot(kpc) + theme_classic2() +
  geom_tile(data=mymap, aes(as.factor(Survey_Period), site_specific), alpha=1, fill="#FDB462") + 
  geom_text(aes(Survey_Period, site_specific, label=is.positive), size=6)  +
  facet_wrap(gene~Unique_ptid, scales="free", ncol=11) + xlab("Survey")
SFig3
ggsave(SFig3, file="~/Desktop/NATURE/ABX_SAMPLES/Manuscript/SupplementaryFigure3.png", device="png", 
       height=4, width=12)
```

note that we also see the persistence of the VIM genes in the nares in subject 15
```{r, fig.height=4}
#let's subset on vim and plot
vim = subset(newdf, gene=="VIM")
vim$Survey_Period = as.numeric(as.character(vim$Survey_Period))

#subset on the people who were sequencing positive
vim = subset(vim, Unique_ptid %in% carbs$Unique_ptid) 
ggplot(vim) + theme_classic2() +
  geom_tile(data=mymap, aes(Survey_Period, site_specific), alpha=1, fill="#FDB462") + 
  geom_line(aes(Survey_Period, site_specific, group=site_specific)) +
  geom_text(aes(Survey_Period, site_specific, label=is.positive), size=6)  +
  facet_wrap(gene~Unique_ptid, scales="free", ncol=11) 

```



### let's look at the correlation of the carbepenamse genes with the various reads.

I tested for the significance of the correlation where R2>0.1. This is a simple linear model looking at the correlation between the Abx reads and the unique reads mapping to each subject. The model is not with not aware of multiple samples from subjects.
 
- KPC. If we do not adjust for multiple corrections, then the association between KPC and Klebsiella (R2=0.38) is significant, while no other taxa have a significant correlation. However, we do not reach significance after making the statistical correction.
- OXA. The correlation with Providencia stuartii (R2=0.23) is significant even after adjusting for multiple testing.
- VIM. This appears to be most highly correlated with the abundance of Morganella (R2=0.92),  Proteus (R2=0.89), Providencia (R2=0.61) and Pseudomonas (R2=0.59). However, there are not enough data points to assess significance and of course we know that the relative abundance of all these species is cross-correlated
- NDM. With 1 sample we cannot do a correlation analysis


```{r, fig.height=12}
#select just the columns of interest in the srst data
srstDF = dplyr::select(srstDF, c("library_id",  "gene",  "allele", "coverage",  "depth",       "diffs", 
                                 "uncertainty", "divergence",
                                 "length", "maxMAF", "clusterid", "seqid"))

#we will read in the phyloseq object without a breadth of coverage restriction
myphy = readRDS("~/Desktop/NATURE/Figure1/ESKAPE_phyloseq_nocoveragethreshold.rds")
myreads = data.frame(otu_table(myphy), tax_table(myphy)) %>%
  reshape::melt(., colnames(tax_table(myphy)))
myreads$library_id = myreads$variable

#merge the taxon abundance with the srst results
newDF = plyr::join(srstDF, myreads)
newDF = subset(newDF, species %in% taxa_of_interest) %>%
  subset(., gene %in% c("VIM_Bla" ,  "KPC_Bla", "OXA_Bla" ))

#specify a model
my.formula<- y ~ x

# plot the correlations
p = ggplot(newDF, aes( value, depth)) + 
  geom_point() + scale_x_log10() + 
  scale_y_log10() + 
  facet_wrap(gene~species, ncol=5) +
  geom_smooth(method = "lm")+ 
  stat_poly_eq(formula = my.formula, aes(label = paste(..eq.label.., ..rr.label.., sep = "~~~")), parse = TRUE)  + 
  geom_point() 
p
#are any of the correlation coefficients greater than 0.2 significant
kpc_kp = subset(newDF, species=="s__Klebsiella pneumoniae" & gene=="KPC_Bla")
cor.test(kpc_kp$depth, kpc_kp$value, alternative="greater",method="pearson")

#are any of the correlation coefficients greater than 0.2 significant
oxa_ps = subset(newDF, species=="s__Providencia stuartii" & gene=="OXA_Bla")
cor.test(oxa_ps$depth, oxa_ps$value, alternative="greater",method="pearson")

```

### Let's generate figure 2
Figure 2: Skin of nursing home residents is a reservoir for C. auris, ESKAPE pathogens and antibiotic resistance genes. a) Recovery of MAGs for species (y-axis) for each subject (x-axis) for skin sites, excluding peri-anal samples. Asterisk indicate subjects for whom MAGs were recovered for each species. Colors map to subjects and species where genomes were at least 50% covered (high) vs. samples and species where coverage was less than 50%. b) Clinical metadata summarizing resident’s history of colonization with resistant Staphylococcus aureus (hxmrsa), ESBL producing enterobacteria (hxesbl), CRPA+ Pseudomonas aeruginosa (hxcrpa), CRE producing enterobacteria, carbapenem resistant Acinetobacter baumannii (hxcrab), or Candida auris (hxcauris). c) Detection of 3 carbapenemase producing genes (VIM, KPC, NDM) identified in clinical microbiology findings or surveillance cultures marked as ‘surveillance’ on x-axis. Detection of antibiotic resistance genes by read mapping in shotgun metagenomic data is marked as positive in ‘sequencing’. Each panel represents a unique subject.

```{r}
Left = plot_grid(Fig2A, rel_heights = 0.5, labels="A")
Right = plot_grid(Fig2B, rel_heights = 1, labels="B")
Fig2 = plot_grid(Left, Right, ncol=2, rel_widths =c(1, 0.5))
Fig2
ggsave(Fig2, file="~/Desktop/NATURE/ABX_SAMPLES/Manuscript/Submission1/Figure2.pdf", device="pdf", height = 7, width = 12)
```

