---
title: "Figure 1: Recovery of MAGs for WHO Critical Priority Pathogens at high relative abundance "
author: "Diana Proctor"
date: "05/17/2022"
output: html_document
---

last updated: `r format(Sys.Date(), format="%B %d %Y")`

***

last updated: `r format(Sys.Date(), format="%B %d %Y")`

*draft: submission to Nature, August 2023*

*Manuscript Title*: Fungal Candida auris and bacterial ESKAPE pathogens:  Human skin as a reservoir for transmissible microbes and antibiotic resistance in nursing homes

*Authors*: Diana M. Proctor 1, Sarah E Sansom2, Clay Deming1, Sean Conlan1, Ryan A Blaustein1,3, Thomas K. Atkins1,4, NISC Comparative Sequencing Program5, Thelma Dangana2, Christine Fukuda2, Lahari Thotapalli2, Heidi H. Kong6, Michael Y. Lin2, Mary K. Hayden2+ and Julia A. Segre 1+

*Affiliations*:

1Microbial Genomics Section, Translational and Functional Genomics Branch, National Human Genome Research Institute, National Institutes of Health, Bethesda, MD 20892, USA

2 Department of Internal Medicine, Division of Infectious Diseases, Rush University Medical Center, Chicago, IL 60612, USA.

3 Current Address: Department of Nutrition and Food Science, University of Maryland, College Park, MD 20742, USA

4 Current Address: Department of Quantitative and Computational Biology, Princeton University, Princeton, NJ

5 NIH Intramural Sequencing Center, National Human Genome Research Institute, National Institutes of Health, Bethesda, MD 20892, USA.

6 Dermatology Branch, National Institute of Arthritis and Musculoskeletal and Skin Diseases, National Institutes of Health, Bethesda, MD 20892, USA


+Contributed equally

---
***
The input for this is the following files, generated using the mags.sh script:

1. *unique.tab files indicating the number of unique reads that map to each mag generated from mags.sh
2. gtdbtk.arcbaceuk_nature.summary_withabx.csv, which is a blend of the GATK taxonomic output (generated from mags.sh) and the mash/mummer output (generated from mags.sh); I manually concatenated these two taxonomy files 
3. mapping_file_2023-07-23.csv, which is the metagenome mapping file
4. a frequencies.csv file, which describes the frequencies generated in code chunk 8
5. final_manifest_genomes.csv, a file which describes the isolate genomes for the study facility D
6. SupplementaryTable1_metagenomesummaries.csv, which describes the sequencing characteristics of the mags
7. SupplementaryTable4_metagenomesummaries.csv, which describes the sequencing characteristics of the shotgun data
8. Cauris_Analytic_2020-2-10.csv, which describes the CFU counts for Candida auris for the samples

This script will generate the following plots:

1. Figure 1
2. Supplementary Figure 1

This output:

1. a phyloseq object including the counts, mapping file, and gtkdb taxonomy for use in subsequent analysis

```{r global_options}
library(knitr)
opts_chunk$set(fig.width=12, fig.height=8, 
               echo=TRUE, warning=FALSE, message=FALSE, error = FALSE)
```


```{r load packages, message=FALSE, warning=FALSE}
 #load package method from from Dan Sprokett

# set seed
set.seed(78979)


#define packages
packages <- c("stringr", 
              "ggplot2", 
              "tidyverse", 
              "ComplexHeatmap", 
              "gridExtra", 
              "fantaxtic", 
              "pals", 
              "reshape2",
              "phyloseq",
              "ggpubr",
              "RColorBrewer",
              "confintr")

# install packages from bioconductor
BiocManager::install(setdiff(packages,installed.packages()), update=FALSE)
n <- length(packages) - sum(sapply(packages,require,character.only=TRUE))

# print if packages loaded properly
if(n == 0){
  print("All necessary R packages loaded properly")
} else {
  print(paste0(n, " R packages did not load properly"))
}
```


let's define some color palettes
```{r}
mycolors=c("Candida auris"= "#8B008B",
      "Pseudomonas aeruginosa"="#C51B7D",
      "Proteus mirabilis"="#FB8072",
      "Providencia stuartii"=  "#FDB462"  ,
      "Escherichia coli"="#737000"   ,
      "Enterococcus faecalis"="#7BE495"  ,
      "Acinetobacter baumannii"="#A1D76A" ,
      "Morganella morganii"="#32612D" ,
      "Klebsiella pneumoniae" = "#0D52BD", 
      "Staphylococcus aureus" = "#329D9C" ,
      "Staphylococcus pettenkoferi" = "gray")


```


define knitr options

```{r}
knitr::opts_chunk$set(echo=TRUE, warning=FALSE, message=FALSE, error = FALSE)
```




read in the *unique.tab files, generated using read mapping, and create a phyloseq object
```{r}
path <- "~/Desktop/NATURE/ABX_SAMPLES/abx" 

#read in the read mapping results
bwaOut <- sort(list.files(path, pattern="*unique.tab", full.names = TRUE))

# Extract sample names
sample.names <- sapply(strsplit(basename(bwaOut), "/"), `[`, 1)
sample.names = stringr::str_remove_all(sample.names, "_mapping_eukBacBins.derep_unique.tab")


#read in the files and store them in a list
TSV.list <- vector("list", length(sample.names))
  for(i in 1:length(sample.names)) {
        df  = data.table::fread(bwaOut[i])
        colnames(df) = c("Genome", "Length", "Counts", "Depth", "Coverage", "Coeff")
        df$Counts_Adjusted_30p = ifelse(df$Coverage <0, 0, df$Counts)
        df = dplyr::select(df,"Genome","Counts_Adjusted_30p")
        colnames(df) = c("Genome", paste0(sample.names[i], "_", "Counts_Adjusted_30p"))
      TSV.list[[i]] =  df
}

#now let's turn this list into a data frame
dt<-rlist::list.cbind(TSV.list)
Genome = dt$Genome
data <- dplyr::select(dt, -Genome)  
data = data.frame(data)
rownames(data) = Genome

#we need to fix column names so they are just metcodes
mynames = stringr::str_remove_all(colnames(data), "_Counts_Adjusted_30p")
colnames(data) = mynames

#fix genome names so that they don't have .fa and can be merged with gtdbk output
rownames(data) = stringr::str_remove_all(rownames(data), ".fa")

#make the otu table
otus = phyloseq::otu_table(data, taxa_are_rows = TRUE)

#make the taxonomy table
#note that  i separated the classificatioon variable in to linnaean tax classes in Excel
#i labeled the tax classes with domain, phylum, etc. all lower case
#i also changed genome to user_genome
tax =read.csv("~/Desktop/NATURE/taxonomy/gtdbtk.arcbaceuk_nature.summary.csv") %>%
  dplyr::select(., c("user_genome", "domain", "phylum", "class", "order", "family", "genus", "species"))

#how many taxa are there per domain of life
#table(NumberPerDomain$domain)

#clean up the taxa names
tax$user_genome = str_remove_all(tax$user_genome, ".fa")
#turn the gtdbk data into a taxonomy table for phyloseq
taxa.names = tax$user_genome
tax = phyloseq::tax_table(as.matrix((tax)))
phyloseq::taxa_names(tax) = taxa.names


#read in the metagenome metadata
map = map = read.csv("~/Desktop/NATURE/mapping_file_2023-07-23.csv") %>%
  subset(., library_id %in% c(sample_names(otus), "Met5675sub"))
map = sample_data(map)
sample_names(map) = map$library_id

#now merge the mapping file, taxonomy table and otu table into phyloseq
phy = phyloseq(otus, sample_data(map), tax) 
colnames(sample_data(phy))[8] = "Label"
#saveRDS(phy, file="~/Desktop/NATURE/Figure1/ESKAPE_phyloseq_nocoveragethreshold.rds")
#saveRDS(phy, file="~/Desktop/NATURE/Figure1/ESKAPE_phyloseq_50percent.rds") 
```


Let's create a grouping variable where we identify the species we're interested in by their species name and label everything else as other.
```{r}
#define the taxa of interest
taxa_of_interest = c("Candida auris" ,
      "s__Escherichia coli",
      "s__Enterococcus faecalis",
     "s__Klebsiella pneumoniae" ,
     "s__Acinetobacter baumannii",
      "s__Pseudomonas aeruginosa",
      "s__Proteus mirabilis",
      "s__Providencia stuartii",
      "s__Staphylococcus pettenkoferi",
     "s__Staphylococcus aureus",
     "s__Morganella morganii"
 )

#create a phyloseq object of the taxa of interest
myTaxa = subset_taxa(phy, species %in% taxa_of_interest)
tax = data.frame(myTaxa@tax_table@.Data)
tax$group = tax$species
tax$group = ifelse(tax$group %in% c(
     "Malassezia arunalokei", 
     "Malassezia globosa CBS 7966", 
     "Malassezia restricta CBS 7877", 
     "Malassezia furfur", 
     "uncultured Malassezia"), "Malassezia", tax$group)

#annotate the taxonomy table for the new phyloseq object defining species of interest     
names = tax$user_genome
tax = as.matrix(tax)
tax_table(myTaxa) = tax



#create a phyloseq object for the species that are NOT in the taxa of interest vector and annotate the tax table
notMyTaxa = subset_taxa(phy, !(species %in% taxa_of_interest))
tax = data.frame(notMyTaxa@tax_table@.Data)
tax$group ="Other"
names = tax$user_genome
tax = as.matrix(tax)
tax_table(notMyTaxa) = tax


#merge the two newly created phyloseq objects so we have one containing a grouping variable
newPhy = merge_phyloseq(notMyTaxa, myTaxa) %>%
  transform_sample_counts(., fun=function(x) 100*(x/sum(x))) %>%
  subset_samples(., !(site_specific %in% c("Ctrl", "Tc", "Bu/To", "Ne", "Ea", "Ax", "Ctrl")))



#Let's collapse the commensal Malassezia species together, but keep sloofiae separate. We'll then create a plot of the species of interest.
#redefine Malassezia as M. spp.
myTaxa = subset_taxa(newPhy, species %in% taxa_of_interest)
tax = data.frame(myTaxa@tax_table@.Data)
tax$group = tax$species
tax$group = ifelse(tax$group %in% c(
     "Malassezia arunalokei", 
     "Malassezia globosa", 
     "Malassezia restricta", 
     "Malassezia furfur", 
     "uncultured Malassezia"), "Malassezia spp.", tax$group)
tax$group = stringr::str_remove_all(tax$group, "s__")
names = tax$user_genome
tax = as.matrix(tax)
tax_table(myTaxa) = tax


notMyTaxa = subset_taxa(newPhy, !(species %in% taxa_of_interest))
tax = data.frame(notMyTaxa@tax_table@.Data)
tax$group ="Other"
tax$group = stringr::str_remove_all(tax$group, "s__")
names = tax$user_genome
tax = as.matrix(tax)
tax_table(notMyTaxa) = tax

#combine the annotated tables
newPhy = merge_phyloseq(notMyTaxa, myTaxa) %>%
  transform_sample_counts(., fun=function(x) 100*(x/sum(x))) 

#fix the body site labels
sample_data(newPhy)$Body_Site = plyr::revalue(sample_data(newPhy)$site_specific, c("An" = "Peri-anal",
                                "Ic"="Inguinal crease", 
                                "N"="Nares",
                                "Fg"="Fingertips",
                                "Tw"= "Toe web"))


#force the ordering of the data frame so it'll plot in the order we want
ordering = c( "Toe web" , "Fingertips" , "Nares", "Ear Canal", "Peri-anal",   "Inguinal crease", "Control")
sample_data(newPhy)$Body_Site <- factor(sample_data(newPhy)$Body_Site, levels = ordering) 

```

let's generate a stacked barplot. it's kind of hard to see anything. we will remove the "other" species
```{r}
 
p = plot_bar(newPhy, x="Label", y="Abundance", fill="group") + 
  geom_bar(aes(color=group, fill=group), stat="identity", position="stack")+
  facet_wrap(~site_specific, scales="free_x", ncol=7) +
  scale_fill_manual(values=mycolors)+ scale_color_manual(values=mycolors)+
  theme_classic()+
         theme(axis.text.x=element_blank(), #remove x axis labels
        axis.ticks.x=element_blank()) +xlab("") + ylab("Relative Abundance")
    
p
```

Let's make Figure 1B: Relative abundance of each of the select species including C. auris and ESKAPE pathogens. Panels, colors, and sample order are as in panel a.
```{r}
#let's get the data out of the previous plot and subset on just our select species
dat = p$data
dat$group2 = ifelse(dat$group %in% "Other", dat$group, "select") 
dat = subset(dat, group !="Other") 

#let's define a variable to order the species by so Candida auris is always on top
ordering = c("Candida auris",
      "Acinetobacter baumannii",
      "Enterococcus faecalis",
      "Escherichia coli" ,
      "Klebsiella pneumoniae", 
       "Morganella morganii",
      "Proteus mirabilis",
      "Providencia stuartii",
      "Pseudomonas aeruginosa",
      "Staphylococcus aureus",
      "Staphylococcus pettenkoferi")

#re-order the data frame
dat$group <- factor(dat$group, levels = ordering)

#make the plot
Fig1B_alt = ggplot(dat, aes(Sample, Abundance, fill=group)) + 
  geom_col(position="stack") + 
  facet_wrap(~Body_Site, scales="free_x", ncol=6) + 
  scale_fill_manual(values=mycolors) + 
  theme(legend.position = "bottom")+ theme_classic()+
  theme(axis.text.x=element_blank(), #remove x axis labels
        axis.ticks.x=element_blank()) + 
  theme(legend.position = "none")  

Fig1B_alt
```

#### Generate Figure 1A

```{r, fig.height=5}
#let's read in the gtdbk data again
gtdbk = read.csv("~/Desktop/NATURE/taxonomy/gtdbtk.arcbaceuk_nature.summary_withabx.csv") %>%
  dplyr::select(., c("user_genome", "msa_percent", "phylum", "genus", "species"))
  #add a variable to merge the sample data to the mag data
  gtdbk$sample = stringr::str_sub(gtdbk$user_genome, 1, 7)

#let's read in the map again
map = read.csv("~/Desktop/NATURE/mapping_file_2023-07-23.csv")
  #add a sample variable so we can merge this with the gdtbk data
  map$sample = map$library_id

#merge the gtdbk and mapping data and then identify the samples that gave rise to MAGs by cross-tabulating 
#we subset the big data table on just the species of interest in figure 1B
df = plyr::join(map, gtdbk)
crossTab = data.frame(table(df$species,df$Sample)) %>%
    subset(., Var1 %in% taxa_of_interest)

#let's recide the Malassezia spp. to one category 
crossTab$group = plyr::revalue(crossTab$Var1, c("Malassezia arunalokei" ="Malassezia spp.", 
     "Malassezia globosa"="Malassezia spp.", 
     "Malassezia restricta"="Malassezia spp.", 
     "Malassezia furfur"="Malassezia spp.",
     "uncultured Malassezia"="Malassezia spp."))

#let's make a denovo mapping file and drop the body sites that don't make it into the main display
newmap = reshape2::colsplit(crossTab$Var2, ";", c("subject", "site", "survey"))
crossTab = data.frame(crossTab, newmap) %>%
  subset(., !(site %in% c("Site_Bu/To" ,"Site_Ne",    "Site_Ctrl", "Site_Tc", "Site_Ea", "Site_Ax", "")))


#let's clean up the body site names
crossTab$Body_Site = plyr::revalue(crossTab$site,
                              c("Site_An" = "Peri-anal",
                                "Site_Ic"="Inguinal crease", 
                                "Site_N"="Nares",
                                "Site_Fg"="Fingertips",
                                "Site_Tw"= "Toe web"))

#let's specify the order of the panels we'd like to use and force the data frame to follow this order
site_ordering = c("Toe web","Fingertips", "Nares","Peri-anal", "Inguinal crease")
crossTab$Body_Site=factor(crossTab$Body_Site, levels = site_ordering) 
crossTab0 = crossTab
crossTab$Freq = ifelse(crossTab$Freq==2, 1, crossTab$Freq)
crossTab$Freq = ifelse(crossTab$Freq==1, 0.5, crossTab$Freq)
#drop this P. aeruginosa that is likely another species
crossTab = subset(crossTab, group !="Pseudomonas aeruginosa_A")
#clean up the species names
crossTab$group = stringr::str_remove_all(crossTab$group, "s__")

Fig1A = ggplot(crossTab, aes(Var2, group, color=group, size=Freq)) + geom_point()+ theme_classic() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size=6)) + 
  facet_wrap(~Body_Site, ncol=7, scales = "free_x") +
  scale_color_manual(values=mycolors) + scale_fill_manual(values=mycolors) +
  theme(legend.position = "none")+
  theme(axis.text.y = element_text(face = "italic"),
        axis.text.x=element_blank(), #remove x axis labels
        axis.ticks.x=element_blank() #remove x axis ticks
          ) + xlab("") + ylab("Species")+
  scale_size_continuous(range = c(0, 1))

Fig1A
```


let's see how many MAGs we recover per sample, for these select species. >60% samples yield a MAG and more than 20% of samples yield 4
```{r, fig.height=4, fig.width=4}
gtdbk$prev = 1

SelectSp = subset(gtdbk, species %in% taxa_of_interest)
PrevBySample = doBy::summary_by(prev~sample, FUN=c(sum, length), data=SelectSp)
PrevBySample$Number = 1:nrow(PrevBySample)
p = ggplot(PrevBySample, aes(prev.sum)) + geom_histogram(bins=20) + theme_classic() 
p

#how many samples yielded at least one MAG for these species?
fm = subset(PrevBySample, prev.sum > 0)
nrow(fm)/266 #266 is the number of non-control samples

#how many samples yielded 4 or more of these select species
fm = subset(PrevBySample, prev.sum >= 4)
nrow(fm)/length(unique(PrevBySample$sample))
```


### Figure 1
Figure 1: Recovery of MAGs for C. auris and ESKAPE genomes and other predominant bacteria and fungi from nursing home microbiome sampling. a) Species recovered are in rows with large points indicating a MAG was recovered. Colors correspond to the species. Body site for sampling is labeled on top. Malassezia spp. encompasses M. restricta, M. arunalokei, M. furfur, M. globosa, and an uncultured Malassezia species. b) Relative abundance of each of the select species including C. auris and ESKAPE pathogens. Panels, colors, and sample order are as in panel a.

```{r}
Figure1 = cowplot::plot_grid(Fig1A, Fig1B_alt, ncol = 1, align = "v", rel_heights = c(1, 0.95), 
  labels=c("A", "B"))
Figure1
ggsave(Figure1, file="~/Desktop/NATURE/ABX_SAMPLES/Manuscript/Submission1/Figure1.pdf", device="pdf", width = 7.5, height = 6)

```

how many of the MAG's had greater than 80% aligned fraction for each species? 
```{r, fig.height=4, fig.width=4}
gtdbk = read.csv("~/Desktop/NATURE/taxonomy/gtdbtk.arcbaceuk_nature.summary_withabx.csv") %>%
  subset(. , species %in% taxa_of_interest) %>%
  subset(., domain != "Fungi")

gtdbk$greaterthan80 = ifelse(gtdbk$msa_percent> 80, 1, 0)

mysummary = doBy::summary_by(greaterthan80~species, FUN=c(sum, length), data=gtdbk)

p = ggplot(mysummary, aes(species, greaterthan80.sum/greaterthan80.length)) + geom_col() + coord_flip()
p
```

what is the breadth and depth of coverage for each of these species


read in the *unique.tab files, generated using read mapping
```{r}
path <- "~/Desktop/NATURE/ABX_SAMPLES/abx" 

# Forward and reverse fastq filenames have format: cat.tsv, the output of the blast searches on the 4 genomes
bwaOut <- sort(list.files(path, pattern="*unique.tab", full.names = TRUE))

# Extract sample names, assuming filenames have format: SAMPLENAME_XXX.fastq
sample.names <- sapply(strsplit(basename(bwaOut), "/"), `[`, 1)
sample.names = stringr::str_remove_all(sample.names, "_mapping_eukBacBins_aureus.derep_unique.tab")

#read in the files and store them in a list
TSV.list <- vector("list", length(sample.names))
  for(i in 2:length(sample.names)) {
        df  = data.table::fread(bwaOut[i])
        colnames(df) = c("Genome", "Length", "Counts", "Depth", "Coverage", "Coeff")
      TSV.list[[i]] =  data.frame(df)
}


data = do.call(rbind, TSV.list)
data$Genome = stringr::str_remove_all(data$Genome, ".fa")

#subset on depth exceeds 1x
covExceeds5 = subset(data, Depth > 5)
colnames(covExceeds5)[1] = "user_genome"

#read in the gtdbk file again and subset on the select species
tax =read.csv("~/Desktop/NATURE/taxonomy/gtdbtk.arcbaceuk_nature.summary.csv") %>%
  dplyr::select(., c("user_genome", "domain", "phylum", "class", "order", "family", "genus", "species")) %>%
  subset(., species %in% c("s__Staphylococcus pettenkoferi" , 
                           "s__Escherichia coli", 
                           "s__Klebsiella pneumoniae",
                          "s__Acinetobacter baumannii" ,  
                          "s__Pseudomonas aeruginosa" , 
                          "Candida auris" ,  
                          "s__Providencia stuartii",
                          "s__Proteus mirabilis"))

#subset the coverage data frame to only include the taxa of interest
covExceeds5 = subset(covExceeds5, user_genome %in% tax$user_genome)
foo = merge(tax, covExceeds5) %>%
  reshape2::melt(., colnames(tax)) %>%
  subset(., variable %in% c("Depth", "Coverage"))

#create a depth data frame
depth = subset(foo, variable=="Depth")
#create a breadth data frame
cov = subset(foo, variable=="Coverage") %>%
  subset(., value > 5)

#define a function to get confidence intervals
get_value_CI <- function(df, species) {
  my_ac = ci_mean(df$value,
      probs = c(0.025, 0.975),
      type =  "bootstrap",
      boot_type = "basic",
      R = 9999,
      seed = 834)
  df = data.frame(CI_lower=my_ac$interval[1], CI_upper=my_ac$interval[2], average=my_ac$estimate, 
                  species=paste0(species))
}
```



### Supplementary Table 5: get summary stats for the depth of coverage 


```{r}
ac = subset(depth, species=="s__Acinetobacter baumannii") %>%
  get_value_CI(., "Acinetobacter baumannii")
ec = subset(depth, species=="s__Escherichia coli") %>%
  get_value_CI(., "Escherichia coli")
kp = subset(depth, species=="s__Klebsiella pneumoniae") %>%
  get_value_CI(., "Klebsiella pneumoniae")
pm = subset(depth, species=="s__Proteus mirabilis") %>%
  get_value_CI(., "Proteus mirabilis")
ps = subset(depth, species=="s__Providencia stuartii") %>%
  get_value_CI(., "Providencia stuartii")
pa = subset(depth, species=="s__Pseudomonas aeruginosa") %>%
  get_value_CI(., "Pseudomonas aeruginosa")
sp = subset(depth, species=="s__Staphylococcus pettenkoferi") %>%
  get_value_CI(., "Staphylococcus pettenkoferi")

depth = data.frame(rbind(ac, ec, kp, pm, ps, pa, sp))
kable(depth)
```


```{r}
#get summary stats for the breath of coverage
ac = subset(cov, species=="s__Acinetobacter baumannii") %>%
  get_value_CI(., "Acinetobacter baumannii")
ec = subset(cov, species=="s__Escherichia coli") %>%
  get_value_CI(., "Escherichia coli")
kp = subset(cov, species=="s__Klebsiella pneumoniae") %>%
  get_value_CI(., "Klebsiella pneumoniae")
pm = subset(cov, species=="s__Proteus mirabilis") %>%
  get_value_CI(., "Proteus mirabilis")
ps = subset(cov, species=="s__Providencia stuartii") %>%
  get_value_CI(., "Providencia stuartii")
pa = subset(cov, species=="s__Pseudomonas aeruginosa") %>%
  get_value_CI(., "Pseudomonas aeruginosa")
sp = subset(cov, species=="s__Staphylococcus pettenkoferi") %>%
  get_value_CI(., "Staphylococcus pettenkoferi")

coverage = data.frame(rbind(ac, ec, kp, pm, ps, pa, sp))
kable(coverage)
```

what fraction of the Candida auris genomes aligned by mummer
```{r}
df = read.csv("~/Desktop/NATURE/ABX_SAMPLES/Manuscript/A.Supplement/archive/SupplementaryTable3_nature_fungal_taxonomy_withABX_v1.1.csv") %>%
  subset(., organism="[Candida] auris") %>%
  subset(., FractionAligned > 0.5)
df$FractionAligned1 = 100*df$FractionAligned 
summary(df$FractionAligned)
```

### supplementary figure 1: Let's make the experimental design figure for the supplement

Supplementary Figure 1: Schematic of samples used for whole genome sequencing (a) and shotgun metagenomics (b). Each point represents a sample. Samples for subjects are shown as a function of time (survey 1-3) with different body sites shown as facets within panels. Abbreviations for body sites are defined as An, Perianal; Fg fingertips/palm; Ic inguinal crease; N nares, and Tw Toe webs. Samples for select body sites are not displayed, including neck sample from subject 3; Bu/To samples from subjects 27, 37, 39; Tc sample from subject 43. 
```{r, fig.height=6}
#define a color palette
mypalette12<-RColorBrewer::brewer.pal(12,"Set3")

#read in the mapping file
metagenomes = read.csv("~/Desktop/NATURE/mapping_file_2023-07-23.csv")
metagenomes$Data_Type = "Shotgun Metagenome"

#how many subjects do we have metagenomic data for
length(unique(metagenomes$Unique_ptid))
#read in the isolate sequencing manifest
genomes = read.csv("~/snps2/cauris_genomes/final_manifest_genomes.csv")


#plot the genomes but only focus on the body sites of interest and the subjects of interest
Sfig1A = subset(genomes, Site %in% c("An", "Ax/Ic",  "Fg", "N",  "Tw", "Ic")) %>%
  subset(., Unique_ptid %in% c("2", "4", "48", "14", "5", "15",
                               "23", "46", "53", "28", "35")) %>%
  subset(., Data_Type=="Isolate") %>%
  ggplot(.,  aes(as.factor(Survey_Period), as.factor(Unique_ptid), color=Data_Type)) + 
  facet_wrap(~Site, ncol=7) + 
  geom_point(size=4) +
  xlab("Survey Period") + 
  ylab("Subject ID") +  
  theme_classic()+ 
  theme(legend.position = "none")


#plot the metagenomes
Sfig1B = metagenomes%>% subset(., site_specific %in% c("An", "Fg", "Ic", "N", "Tw")) %>%
     ggplot(.,aes(as.factor(Survey_Period), as.factor(Unique_ptid), color=Data_Type)) + 
  facet_wrap(~site_specific, ncol=7) + 
  geom_point(size=4, alpha=0.9) +
  xlab("Survey Period") + 
  ylab("Subject ID") + 
  theme_classic() + 
  theme(legend.position = "none")

cowplot::plot_grid(Sfig1A, Sfig1B, ncol = 1, rel_heights = c(1, 2), labels = "auto")
ggsave(cowplot::plot_grid(Sfig1A, Sfig1B, ncol = 1, rel_heights = c(1, 2), labels = "auto"), 
       file="~/Desktop/NATURE/ABX_SAMPLES/Manuscript/SupplementaryFigure1.png", width = 11, height = 8, device="png")
```


look to see if we get candida auris MAGs from high cfu count samples

- 18 of the MAGs were generated from samples with CFU counts at the limit of detection (i.e., 3600 CFU/mL) while 1 sample each yielded MAGs when CFUs ranged from 12 to 36, 168, and 2364. Samples from subjects 16, 28, and 49 were assocaited with the lower CFU count samples
```{r, fig.height=4, fig.width=4}
#subset the gtdbk file on Candida auris and merge withe MAG data with the sample data so we know which samples gave rise to cauris mags
auris = subset(tax, species == "Candida auris")
auris$library_id = stringr::str_sub(auris$user_genome, 1, 7)
aurisDF = merge(auris, map)

#let's read in the CFU data for Canddia auris and clean it up
clin= read.csv("~/Desktop/NATURE/Figure1/Cauris_Analytic_2020-2-10.csv") %>%
  dplyr::select(., c("Unique_ptid", "Survey_Period", "CFUTransform_Cauris", "site"))
clin$is.missing = is.na(clin$CFUTransform_Cauris)
clin = subset(clin, is.missing ==FALSE)
clin$site = as.factor(clin$site)
clin$site_specific = plyr::revalue(clin$site, c("0"="N",
                                                "1"="Ea",
                                                "2" ="Ne",
                                                "3"="Ax",
                                                "4"="Ic",
                                                "5"="An",
                                                "6"="Tw",
                                                "7"="Fg",
                                                "8"="Bu",
                                                "9"="Tc"))

aurisDF2 = merge(clin, aurisDF)
#plot the cauris cfus for the samples that gave rise to mags
p = ggplot(aurisDF2, aes(CFUTransform_Cauris)) + geom_histogram() + theme_classic()
p

aurisDF2$is.high = ifelse(aurisDF2$CFUTransform_Cauris ==3600, "high", "low")
#test to see if MAGs for Cauris assembled out of samples with high sequencing depth
chisq.test(table(aurisDF2$CFUTransform_Cauris))
chisq.test(table(aurisDF2$is.high))
```


how many reads are microbial based on body site?
```{r, fig.height=6}
dat = read.csv("~/Desktop/NATURE/ABX_SAMPLES/Manuscript/A.Supplement/SupplementaryTable1_metagenomesummaries.csv")

p1 = ggplot(dat, aes(site_specific, UNALIGNED.PERCENT )) + geom_boxplot() + geom_point()
 
p2 = ggplot(dat, aes(site_specific, as.numeric(UNALIGNED.READS))) + geom_boxplot() + geom_point() + scale_y_log10()
grid.arrange(p1, p2, ncol=2)
        
```

