---
title: 'Figure 2: the skin is a reservoir for antibiotic resistance'
author: "Diana Proctor"
date: "2023-04-06"
output:
  html_document: default
  pdf_document: default
---
***

last updated: `r format(Sys.Date(), format="%B %d %Y")`

*draft: submission to Nature, August 2023*

*Manuscript Title*: Fungal Candida auris and bacterial ESKAPE pathogens:  Human skin as a reservoir for transmissible microbes and antibiotic resistance in nursing homes

*Authors*: Diana M. Proctor 1, Sarah E Sansom2, Clay Deming1, Sean Conlan1, Ryan A Blaustein1,3, Thomas K. Atkins1,4, NISC Comparative Sequencing Program5, Thelma Dangana2, Christine Fukuda2, Lahari Thotapalli2, Heidi H. Kong6, Michael Y. Lin2, Mary K. Hayden2+ and Julia A. Segre 1+

*Affiliations*:

1Microbial Genomics Section, Translational and Functional Genomics Branch, National Human Genome Research Institute, National Institutes of Health, Bethesda, MD 20892, USA

2 Department of Internal Medicine, Division of Infectious Diseases, Rush University Medical Center, Chicago, IL 60612, USA.

3 Current Address: Department of Nutrition and Food Science, University of Maryland, College Park, MD 20742, USA

4 Current Address: Department of Quantitative and Computational Biology, Princeton University, Princeton, NJ

5 NIH Intramural Sequencing Center, National Human Genome Research Institute, National Institutes of Health, Bethesda, MD 20892, USA.

6 Dermatology Branch, National Institute of Arthritis and Musculoskeletal and Skin Diseases, National Institutes of Health, Bethesda, MD 20892, USA


+Contributed equally

---
***

Make Figure 2 and supplementary figures 2-4.

First let's load needed R packages.
```{r, echo=FALSE}
#set global knitting options
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE, error = FALSE, fig.width = 12, fig.height = 7)

# load package method from from Dan Sprokett

# set seed
set.seed(78979)
#define packages
packages <- c("knitr", 
              "vcfR", 
              "ANCOMBC",
              "ggplot2", 
              "tidyverse", 
              "colorspace", 
              "kableExtra", 
              "gridExtra", 
              "scales",
              "refGenome",
              "RColorBrewer",
              "ape",
              "ggtree",
              "ape",
              "Biostrings",
              "vcfR",
              "adegenet",
              "poppr",
              "pegas",
              "dendextend",
              "cowplot",
              "ggpubr",
              "ComplexHeatmap",
              "phyloseq",
              "reshape2",
              "viridis",
              "ggpmisc")
# install packages from bioconductor
BiocManager::install(setdiff(packages,installed.packages()), update=FALSE)
n <- length(packages) - sum(sapply(packages,require,character.only=TRUE))

# print if packages loaded properly
if(n == 0){
  print("All necessary R packages loaded properly")
} else {
  print(paste0(n, " R packages did not load properly"))
}
```

define the taxa of interest for the figure
```{r}
#define the taxa of interest
taxa_of_interest = c("Candida auris" ,
      "s__Escherichia coli",
      "s__Klebsiella pneumoniae" ,
     "s__Acinetobacter baumannii",
      "s__Pseudomonas aeruginosa",
      "s__Proteus mirabilis",
      "s__Providencia stuartii",
     "s__Morganella morganii",
     "s__Staphylococcus aureus")

bacteria_of_interest = c("s__Escherichia coli",
     "s__Klebsiella pneumoniae" ,
     "s__Acinetobacter baumannii",
      "s__Pseudomonas aeruginosa",
      "s__Proteus mirabilis",
      "s__Providencia stuartii",
     "s__Morganella morganii",
     "s__Staphylococcus aureus")

species_order = c("Escherichia coli","Staphylococcus aureus" ,
                  "Klebsiella pneumoniae" , "Acinetobacter baumannii", 
                  "Pseudomonas aeruginosa" ,"Morganella morganii",     
                  "Proteus mirabilis" ,      "Providencia stuartii"   )

```

### Let's read in the data that we will use
Specifically, let's read in the results of the bwa mapping
We will remove the peri-anal samples
```{r}
df_complete = read.csv("~/Desktop/NATURE/ABX_SAMPLES/Manuscript/Submission2/05_gtdbk/gtdbtk.arcbacfun120.summary_2024-06-26_v2.0.csv")

df = read.csv("~/Desktop/NATURE/ABX_SAMPLES/Manuscript/Submission2/05_gtdbk/gtdbtk.arcbacfun120.summary_2024-06-26_v2.0.csv") %>%
  subset(., species %in% bacteria_of_interest) %>%
  subset(., Unique_ptid != "ZymoD6306") %>%
  subset(., !(site_specific %in% c("Ctrl",  "Bu/To", "An")))

#what subjects did we recover target species mags for: 7, 32, 39, 43
s7 = subset(df_complete, Unique_ptid==7) #commensals
s32 = subset(df_complete, Unique_ptid==32) #petten
s39 = subset(df, Unique_ptid==39) #peri-anal only
s43 = subset(df, Unique_ptid==43) #peri-anal only


#how many subjects did we recover MAGs from
taxa = read.csv("~/Desktop/NATURE/ABX_SAMPLES/Manuscript/Submission2/05_gtdbk/gtdbtk.arcbacfun120.summary_2024-06-26_v2.0.csv") %>%
  subset(., Unique_ptid != "ZymoD6306")
length(unique(taxa$Unique_ptid))
```



### Let's make figure 2c: how persistent is mag recovery over time for each subject, excluding peri-anal sites

C) Recovery of MAGs from skin sites (inguinal crease, fingertips, toe webs, and axilla) and nares over monthly surveillance (x-axis) for the 18 residents who had complete time series. Colors correspond to species, as in Figure 1.
```{r}
complete_subjects = c(12, 14, 15, 16, 18, 2, 23, 26, 27, 28, 3, 31, 33, 35, 4, 40,  53, 54) 

noa = subset(df, species != "s__Staphylococcus aureus")
noa$presence = 1
fm = subset(noa, Unique_ptid %in% complete_subjects)

#clean up the species names
fm$species = stringr::str_remove_all(fm$species, "s__")

#reorder the subjects
myorders = 1:57
fm$Unique_ptid = factor(fm$Unique_ptid, levels=myorders)
fm$species = factor(fm$species, levels=species_order)
Fig2C_colors =c(
      "Pseudomonas aeruginosa"="#C51B7D",
      "Proteus mirabilis"="#FB8072",
      "Providencia stuartii"=  "#FDB462"  ,
      "Escherichia coli"="#737000"   ,
      "Acinetobacter baumannii"="#A1D76A" ,
      "Morganella morganii"="#32612D" ,
      "Klebsiella pneumoniae" = "#0D52BD")
Fig2C = ggplot(fm, aes(Survey_Period, species, color=species)) + geom_point(size=3) + 
  facet_wrap(~Unique_ptid, nrow = 2) + 
  theme_bw() +
  theme(axis.text.y = element_text(face="italic")) + 
  ylab("") +
  xlab("Survey Period") + scale_color_manual(values=Fig2C_colors) +
  theme(legend.position = "none") +
  scale_y_discrete(limits = rev) 
  
Fig2C
```



### now let's determine which samples/subjects were associated with MAGs of each of the select species.

```{r, fig.height=5, fig.width=5}

df_complete = read.csv("~/Desktop/NATURE/ABX_SAMPLES/Manuscript/Submission2/05_gtdbk/gtdbtk.arcbacfun120.summary_2024-06-26_v2.0.csv") %>%
  subset(., species %in% taxa_of_interest) %>%
  subset(., Unique_ptid != "ZymoD6306") %>%
  subset(., !(site_specific %in% c("Ctrl",  "Bu/To", "An")))

#let's cross tabulate the taxonomy and sample information for the species of interest
df$Sample = paste0(df$Unique_ptid, ";", df$site_specific, ";", df$Survey_Period)
crossTab = data.frame(table(df$species,df$Sample)) %>%
    subset(., Var1 %in% taxa_of_interest)


#clean up the variables by removing s__ from the species names
crossTab$group = stringr::str_remove_all(crossTab$Var1 , "s__")

 
#we need to correct for the fact that we have spades assemblies AND megahit assemblies for a select number of samples
#replace any count equal to two with a 1
crossTab$Freq = ifelse(crossTab$Freq==0, 0, 1)
map = data.frame(colsplit(crossTab$Var2, ";",c("Subject", "Site", "Survey")))
crossTab =data.frame(crossTab, map)

#let's now look at how many times we assembled each of the select species out of samples for each subject. We will normalize by the number of samples for each subject.
prevDF = doBy::summary_by(crossTab, Freq~Var1+Subject, FUN=c(sum, length)) 

#now make a bar of prevalence
burden = prevDF
burden$Prev = ifelse(burden$Freq.sum==0, 0, 1)
burden2 = doBy::summary_by(burden, Prev~Subject, FUN=c(sum, length)) 

ordering = c(1:57)
burden2$Unique_ptid <- factor(burden2$Subject, levels = ordering)

#what is the median number of eskape pathogens carried
summary(burden2$Prev.sum)

```

### Supplementary Figure 3

do we see co-occurrence of MAG recovery

Supplementary Figure 3: Probability of recovering MAGs from pairs of species based on presence/absence matrix. Yellow indicates species pairs that were found to co-occur less frequently than expected by chance (p<0.05); blue indicates species pairs that were found to co-occur more frequently than expected by chance (p<0.05); gray indicates species pairs for which a negative or positive association could not be determined.  P. stuartii, P. mirabilis, and K. pneumoniae were detected together more frequently than expected by chance and tended to co-occur with gut commensals such as Bifidobacterium breve and Bacteroides fragilis. In contrast, Staphylococcus pettenkoferi tended to co-occur with Staphylococcus and Corynebacterium species while C. auris tended to co-occur with both P. mirabilis, and P. stuartii as well as S. pettenkoferi, indicating possible gut and skin seeding.
```{r}
library(cooccur)
df_complete = read.csv("~/Desktop/NATURE/ABX_SAMPLES/Manuscript/Submission2/05_gtdbk/gtdbtk.arcbacfun120.summary_2024-06-26_v2.0.csv") 
commenals_list = read.csv("~/Desktop/NATURE/ABX_SAMPLES/Manuscript/Submission2/commensals.csv")
commenals_list = commenals_list$species
mytaxa = c("Candida auris", "s__Escherichia coli",   "s__Klebsiella pneumoniae",   "s__Acinetobacter baumannii",
  "s__Pseudomonas aeruginosa",  "s__Proteus mirabilis",       "s__Providencia stuartii",    
  "s__Morganella morganii",
  "s__Staphylococcus pettenkoferi", "s__Staphylococcus epidermidis", "s__Cutibacterium acnes",  
  "Malassezia arunalokei",   "Malassezia furfur", "Malassezia globosa CBS 7966"  ,    
  "Malassezia restricta CBS 7877",        
  "Malassezia slooffiae",                 commenals_list)

#let's cross tabulate the taxonomy and sample information for the species of interest
df0 = subset(df_complete, species %in% mytaxa)
df0$Sample = paste0(df0$Unique_ptid, ";", df0$site_specific, ";", df0$Survey_Period)
crossTab = data.frame(table(df0$species,df0$Sample))


#clean up the variables by removing s__ from the species names
crossTab$group = stringr::str_remove_all(crossTab$Var1 , "s__")

 
#we need to correct for the fact that we have spades assemblies AND megahit assemblies for a select number of samples
#replace any count equal to two with a 1
crossTab$Freq = ifelse(crossTab$Freq==0, 0, 1)
map = data.frame(colsplit(crossTab$Var2, ";",c("Subject", "Site", "Survey")))
crossTab =data.frame(crossTab, map)

#make a matrix
M = acast(crossTab, Var1~Var2, value.var="Freq")

#run co-occurence analysis
out = cooccur::cooccur(M, spp_names=TRUE, true_rand_classifier=0.1)
plot(out, plotrand = TRUE)

```

### Fig2A 
 A) Presence of species (y-axis) for each resident (x-axis) on skin (inguinal crease, fingertips, toe webs, and axilla) and nares. Presence is defined in two ways. First, an asterisk indicates residents for whom MAGs were recovered for each species. Second, colors indicate residents for whom genomes were > 50% (present, navy blue color) or <50% (absent, gray color) covered in at least one sample, based on read mapping. We use this second definition of presence since if a genome is at least 50% covered, but does not yield a MAG, there is a high likelihood of that species being present. 

```{r}
 convert2_binary = function(phy){
  M = as.matrix(otu_table(phy)) #taxa are rows
  M = vegan::decostand(as.matrix(data.frame(M)), "pa")
  otus = otu_table(M, taxa_are_rows = TRUE)
  otu_table(phy) = otus
  return(phy)
}

#let's read in the results of the read mapping (unique reads) obtained by bwa-mem
#in order for an abundance value to be present in this table, that species must have had at least 50% of its genome covered in that sample
phy = readRDS("~/Desktop/NATURE/ABX_SAMPLES/Manuscript/Submission2/figures/ESKAPE_phyloseq_50percent_missing.rds") %>%
  subset_samples(., site_specific !=  "An")  %>%
  subset_taxa(., species %in% taxa_of_interest)  %>%
  subset_samples(., library_id != "Met5656") %>%
  convert2_binary(.) %>%
  merge_samples(., "Unique_ptid", fun=sum)

  
#make a data frame of the phyloseq object and clean it up
HighCov = data.frame(t(otu_table(phy)) , tax_table(phy))
HighCovDF = melt(HighCov, colnames(tax_table(phy))) %>%
  dplyr::select(., c("species", "variable", "value"))

colnames(HighCovDF) = c("Var1", "Unique_ptid", "Abundance")
HighCovDF$Unique_ptid = stringr::str_remove_all(HighCovDF$Unique_ptid , "X")
HighCovDF$subject = paste0("Subject_", HighCovDF$Unique_ptid )

HighCovDF$HighCov = ifelse(HighCovDF$Abundance > 0, 1, 0)
HighCovDF = dplyr::select(HighCovDF, "Unique_ptid", "Var1", "HighCov")
HighCovDF$HighCov =  ifelse(HighCovDF$HighCov > 0, ">50%", "<50%")
HighCovDF$Unique_ptid = as.numeric(as.character(HighCovDF$Unique_ptid))



#merge the coverage information with the prevalence information generated in the previous chunk
PA = prevDF
PA$prev =  ifelse(PA$Freq.sum > 0, "Yes", "No")
PA$prev = as.factor(PA$prev)
PA_DF = PA %>%
  dplyr::select(., c("Subject", "Var1", "prev"))



PA_DF$Unique_ptid = as.numeric(as.character(PA_DF$Subject))


#merge coverage and presence/absence
FinalDF = merge(PA_DF, HighCovDF) 
FinalDF$Var1 = stringr::str_remove_all(FinalDF$Var1, "s__")
FinalDF$MAG = FinalDF$prev
#FinalDF = subset(FinalDF, Var1 !="0")


#make the plot
ordering = c(1:60)
FinalDF$Unique_ptid <- factor(FinalDF$Unique_ptid, levels = ordering)
FinalDF = subset(FinalDF, Var1 != "Candida auris")


#create an annotation data frame for whether or not we recovered a mag for a sample
annotation = subset(FinalDF, MAG=="Yes")
annotation$annotation = "*"
newcolours=c(">50%"= "midnightblue", "<50%"= "lightskyblue4" ) 

#make figure 2A
FinalDF$Var1 = factor(FinalDF$Var1, levels=species_order)
Fig2A = ggplot(FinalDF, aes(as.factor(Unique_ptid), Var1))+ 
  geom_tile(aes(fill=HighCov, color=HighCov)) + 
  geom_text(data=annotation, aes( x=as.factor(Unique_ptid), y=Var1, label=annotation),color="white", fontface="bold" ) + 
  scale_fill_manual(values=newcolours, name = "Legend") +
  scale_color_manual(values=newcolours, name = "Legend") +
  theme_classic() + xlab("Subject") + 
  theme(axis.text.y = element_text(face = "italic")) + ylab("") + 
  theme(legend.position="bottom") +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  scale_y_discrete(limits = rev)
Fig2A
```


how many people had Ecoli MAGs vs. 50% coverage
```{r}
ec = subset(FinalDF, Var1=="Escherichia coli")
ec_cov = subset(ec, HighCov==">50%") %>%
  subset(., MAG=="No")
length(unique(ec_cov$Unique_ptid))

ec = subset(FinalDF, Var1=="Escherichia coli") %>%
  subset(., MAG=="Yes")
length(unique(ec$Unique_ptid))
```



### Figure 2B
B) Detection of 3 carbapenemase genes (bla-KPC, bla-VIM, bla-NDM) identified in historical public health surveillance reports are marked as ‘surveillance’ on the top panel. Detection of antibiotic resistance genes within skin microbiome by read mapping in shotgun metagenomic data are marked as ‘sequencing’ on the lower panel. Resident is on the x-axis and the carbapenemase gene is on the y-axis, with shape corresponding to each gene. 

okay, now let's plot the antibiotic resistance data, derived from srst2, a read-based approach.

```{r, fig.height=8, fig.width=6}
map = read.csv("~/Desktop/NATURE/ABX_SAMPLES/Manuscript/Submission2/figures/SupplementaryTable2_metagenomesummaries_v3.csv") %>%
  subset(., !(site_specific %in% c( "Ctrl", "Bu/To" , "ZymoD6306")))

#read srst2 in
srst = read.csv("~/Desktop/NATURE/ABX_SAMPLES/Manuscript/Submission2/10_srst2/srst2_full_results_batch3_combined.csv") %>%
  subset(., Sample %in% map$library_id)
colnames(srst)[1] = "library_id"

#merge srst2 with the sample metadata and drop samples
srstDF = plyr::join(srst, map, by="library_id") 
#write.csv(srstDF,"~/Desktop/NATURE/ABX_SAMPLES/Manuscript/Submission2/figures/srstDF_annotated_withabx.csv")


#define the genes we're interested in seeing
Blas2 = c( "KPC_Bla", "VIM_Bla" ,  "NDM_Bla")

#subset the srst2 results on just the carbepenamse genes since we have surveillance data for them
carbs = subset(srstDF, gene %in% Blas2) 
carbs$gene = stringr::str_sub(carbs$gene, 1, 3)
carbs = subset(carbs, gene %in% c("KPC", "NDM", "VIM"))
carbs = dplyr::select(carbs, c("Unique_ptid", "Survey_Period", "site_specific", "gene"))

#write.csv(carbs, file="carb_sequencing_positive.csv")
#how many subjects have these genes?
kpc = subset(carbs, gene=="KPC")
#write.csv(kpc, file="kpc_seqpositive.csv")

#create a variable in the sequencing results to denote positivity and identification type
carbs$is.positive = "+"
carbs$Type = "Sequencing"
SeqPositiveSubjects = unique(carbs$Unique_ptid)


#read in the surveillance/clinical microbiology findings
PCRpositives = read.csv("~/Desktop/NATURE/ABX_SAMPLES/Manuscript/Submission2/figures/SupplementaryTable6.csv") 

#subset on the findings with a mechanism of action and annotate
PCRpositives = dplyr::select(PCRpositives, c("Unique_ptid","Mechanism", "Sample_type"))
colnames(PCRpositives)[3] = "site_specific"
PCRpositives$Survey_Period = ""
colnames(PCRpositives)[2] = "gene"
PCRpositives$is.positive = "+"
PCRpositives$Type = "Surveillance"

PCRpositives = subset(PCRpositives, gene != "Unknown")
#make a data frame combining sequencing results and the surveillance findings and subset on the patients we analyzed by shotgun
mydf = data.frame(rbind(PCRpositives, carbs)) %>%
  subset(., Unique_ptid %in% SeqPositiveSubjects)


#order the data frame for plotting
ordering = c("Surveillance", "Sequencing"  )
mydf$Type <- factor(mydf$Type, levels = ordering)
myorder = as.factor(1:57)
mydf$Subject = factor(mydf$Unique_ptid, levels=myorder)
#write.csv(newdf, file="~/Desktop/NATURE/ABX_SAMPLES/surveillance_sequencing.csv")

#plot
myorder = 1:57
mydf$Subject = factor(mydf$Subject, levels=myorder)
mydf$Gene = mydf$gene


Fig2B_Alt = ggplot(mydf, aes(Subject, Gene, label=is.positive, shape=Gene)) + 
  facet_wrap(~Type,  strip.position="left", ncol=1) + 
  geom_point(size=3)+
   ylab("")+ 
   xlab("Subject ID") + 
  theme_bw() +
  theme(strip.background = element_blank(),
    strip.placement = "outside",
    strip.text.y = element_text(size=10, face="bold", angle=0),
     strip.text.x = element_text(size=10, face="bold", angle=0)) + 
  theme(axis.text = element_text(size = 10))

```

### Let's generate figure 2
Figure 2: Skin of nursing home residents is a reservoir for C. auris, ESKAPE pathogens and antibiotic resistance genes. A) Presence of species (y-axis) for each resident (x-axis) on skin (inguinal crease, fingertips, toe webs, and axilla) and nares. Presence is defined in two ways. First, an asterisk indicates residents for whom MAGs were recovered for each species. Second, colors indicate residents for whom genomes were > 50% (present, navy blue color) or <50% (absent, gray color) covered in at least one sample, based on read mapping. We use this second definition of presence since if a genome is at least 50% covered, but does not yield a MAG, there is a high likelihood of that species being present. B) Detection of 3 carbapenemase genes (bla-KPC, bla-VIM, bla-NDM) identified in historical public health surveillance reports are marked as ‘surveillance’ on the top panel. Detection of antibiotic resistance genes within skin microbiome by read mapping in shotgun metagenomic data are marked as ‘sequencing’ on the lower panel. Resident is on the x-axis and the carbapenemase gene is on the y-axis, with hsape corresponding to each gene. C) Recovery of MAGs from skin sites (inguinal crease, fingertips, toe webs, and axilla) and nares over monthly surveillance (x-axis) for the 18 residents who had complete time series. Colors correspond to species, as in Figure 1.

```{r}
Left = plot_grid(Fig2A, rel_heights = 0.25, labels="A")
Right = plot_grid(Fig2B_Alt, rel_heights = 0.5, labels="B")
Bottom = plot_grid(Fig2C, rel_heights = 1, labels="C")
Top = plot_grid(Left,Right, ncol=2, rel_widths =c(1, 0.75))
Fig2 = plot_grid(Top,Bottom, ncol=1, rel_heights = c(1, 0.75))
ggsave(Fig2, file="~/Desktop/NATURE/ABX_SAMPLES/Manuscript/Submission2/figures/Figure2.pdf", device="pdf", height = 7, width = 9)
```

### SFig5: how persistent are these genes over time

Supplementary Figure 5: Detection of bla-KPC at multiple time points across subjects and skin sites and nares. Each panel represents a subject. Body sites are displayed on the y-axis and time point is displayed on the x-axis. Orange boxes are displayed where a sample was sequenced for that subject, site, and time point. Crosses are displayed in cases where bla-KPC was detected by SRST2. Persistent detection of bla-KPC is observed in subject 14, 23, 31, 53, and 54 with potential persistence observed in subjects 27 and 46. Persistence is defined as the detection of a gene at consecutive monthly surveys. Tw, toe webs; N, nares; Ic, inguinal crease; Fg, Fingertips/palm; An, peri-anus.
```{r, fig.height=4}
mymap = subset(map, Unique_ptid %in% carbs$Unique_ptid)
mymap = subset(mymap, Unique_ptid !=15)
newdf = subset(mydf, Type=="Sequencing")

#let's subset on KPC and plot
kpc = subset(newdf, gene=="KPC")
kpc$Survey_Period = as.numeric(as.character(kpc$Survey_Period))

#subset on the people who were sequencing positive
kpc = subset(kpc, Unique_ptid %in% carbs$Unique_ptid) 
kpc = subset(kpc, Unique_ptid != 15)
myorder = as.factor(1:57)
kpc$Subject = factor(kpc$Unique_ptid, levels=myorder)
SFig5 = ggplot(kpc) + theme_classic() +
  geom_tile(data=mymap, aes(as.factor(Survey_Period), site_specific), alpha=1, fill="#FDB462") + 
  geom_text(aes(Survey_Period, site_specific, label=is.positive), size=6)  +
  facet_wrap(gene~Unique_ptid, scales="free", ncol=6) + 
  xlab("Survey")

SFig5
ggsave(SFig5, file="~/Desktop/NATURE/ABX_SAMPLES/Manuscript/Submission2/figures/SupplementaryFigure5.png", device="png", 
       height=4, width=12)
```

vim
```{r}
mymap = subset(map, Unique_ptid %in% carbs$Unique_ptid)
newdf = subset(newdf, Type=="Sequencing") %>%
  subset(., site_specific %in% c("N", "Ic", "Tw", "An", "Fg"))

#let's subset on VIM and plot
VIM = subset(newdf, gene=="VIM")
VIM$Survey_Period = as.numeric(as.character(VIM$Survey_Period))

#subset on the people who were sequencing positive
VIM = subset(VIM, Unique_ptid %in% carbs$Unique_ptid) 
myorder = as.factor(1:57)
VIM$Subject = factor(VIM$Unique_ptid, levels=myorder)


SFig3b = ggplot(VIM) + theme_classic() +
  geom_tile(data=mymap, aes(as.factor(Survey_Period), site_specific), alpha=1, fill="#FDB462") + 
  geom_text(aes(Survey_Period, site_specific, label=is.positive), size=6)  +
  facet_wrap(gene~Subject, scales="free", ncol=6) + 
  xlab("Survey")

SFig3b
```

#### Supplementary Figure 4: Look at only the mags that are at least 90% complete

Supplementary Figure 4: Presence of species (y-axis) for each resident (x-axis) on skin (inguinal crease, fingertips, toe webs, and axilla) and nares. Presence is defined in two ways. First, an asterisk indicates residents for whom MAGs which were at least 90% complete were recovered for each species, as defined by > 90% checkM2 completeness and > 90% aligned. Second, colors indicate residents for whom genomes were > 50% (present, navy blue color) or <50% (absent, gray color) covered in at least one sample, based on read mapping. We use this second definition of presence since if a genome is at least 50% covered, but does not yield a MAG, there is a high likelihood of that species being present. Subjects 7, 32, 39 and 43 are not shown, as no MAGs from skin sites other than peri-anal skin were recovered for these subjects.
```{r}
df_complete = read.csv("~/Desktop/NATURE/ABX_SAMPLES/Manuscript/Submission2/figures/SupplementaryTable3_all_annotated_passonly_nodups_assembler.csv")  %>%
  subset(., is.failure != "yes")
df = read.csv("~/Desktop/NATURE/ABX_SAMPLES/Manuscript/Submission2/figures/SupplementaryTable3_all_annotated_passonly_nodups_assembler.csv") %>%
  subset(., species %in% taxa_of_interest) %>%
  subset(., Unique_ptid != "ZymoD6306") %>%
  subset(., !(site_specific %in% c("Ctrl",  "Bu/To", "An"))) %>%
  subset(., Completeness > 90) %>%
  subset(., msa_percent > 90) %>%
  subset(., is.failure != "yes")

#let's cross tabulate the taxonomy and sample information for the species of interest
df0 = subset(df, species %in% mytaxa)
crossTab = data.frame(table(df0$species,df0$Sample))


#clean up the variables by removing s__ from the species names
crossTab$group = stringr::str_remove_all(crossTab$Var1 , "s__")

 
#we need to correct for the fact that we have spades assemblies AND megahit assemblies for a select number of samples
#replace any count equal to two with a 1
crossTab$Freq = ifelse(crossTab$Freq==0, 0, 1)
map = data.frame(colsplit(crossTab$Var2, ";",c("Subject", "Site", "Survey")))
crossTab =data.frame(crossTab, map)


 convert2_binary = function(phy){
  M = as.matrix(otu_table(phy)) #taxa are rows
  M = vegan::decostand(as.matrix(data.frame(M)), "pa")
  otus = otu_table(M, taxa_are_rows = TRUE)
  otu_table(phy) = otus
  return(phy)
}

#let's read in the results of the read mapping (unique reads) obtained by bwa-mem
#in order for an abundance value to be present in this table, that species must have had at least 50% of its genome covered in that sample
phy = readRDS("~/Desktop/NATURE/ABX_SAMPLES/Manuscript/Submission2/figures/ESKAPE_phyloseq_50percent_missing.rds") %>%
  subset_samples(., site_specific !=  "An")  %>%
  subset_taxa(., species %in% taxa_of_interest)  %>%
  subset_samples(., library_id != "Met5656") %>%
  convert2_binary(.) %>%
  merge_samples(., "Unique_ptid", fun=sum)

  
#make a data frame of the phyloseq object and clean it up
HighCov = data.frame(t(otu_table(phy)) , tax_table(phy))
HighCovDF = melt(HighCov, colnames(tax_table(phy))) %>%
  dplyr::select(., c("species", "variable", "value"))

colnames(HighCovDF) = c("Var1", "Unique_ptid", "Abundance")
HighCovDF$Unique_ptid = stringr::str_remove_all(HighCovDF$Unique_ptid , "X")
HighCovDF$subject = paste0("Subject_", HighCovDF$Unique_ptid )

HighCovDF$HighCov = ifelse(HighCovDF$Abundance > 0, 1, 0)
HighCovDF = dplyr::select(HighCovDF, "Unique_ptid", "Var1", "HighCov")
HighCovDF$HighCov =  ifelse(HighCovDF$HighCov > 0, ">50%", "<50%")
HighCovDF$Unique_ptid = as.numeric(as.character(HighCovDF$Unique_ptid))



#merge the coverage information with the prevalence information generated in the previous chunk
PA = prevDF
PA$prev =  ifelse(PA$Freq.sum > 0, "Yes", "No")
PA$prev = as.factor(PA$prev)
PA_DF = PA %>%
  dplyr::select(., c("Subject", "Var1", "prev"))



PA_DF$Unique_ptid = as.numeric(as.character(PA_DF$Subject))


#merge coverage and presence/absence
FinalDF = merge(PA_DF, HighCovDF) 
FinalDF$Var1 = stringr::str_remove_all(FinalDF$Var1, "s__")
FinalDF$MAG = FinalDF$prev
#FinalDF = subset(FinalDF, Var1 !="0")


#make the plot
ordering = c(1:60)
FinalDF$Unique_ptid <- factor(FinalDF$Unique_ptid, levels = ordering)
FinalDF = subset(FinalDF, Var1 != "Candida auris")


#create an annotation data frame for whether or not we recovered a mag for a sample
annotation = subset(FinalDF, MAG=="Yes")
annotation$annotation = "*"
newcolours=c(">50%"= "midnightblue", "<50%"= "lightskyblue4" ) 

#makeSFig4
FinalDF$Var1 = factor(FinalDF$Var1, levels=species_order)
SFig4 = ggplot(FinalDF, aes(as.factor(Unique_ptid), Var1))+ 
  geom_tile(aes(fill=HighCov, color=HighCov)) + 
  geom_text(data=annotation, aes( x=as.factor(Unique_ptid), y=Var1, label=annotation),color="white", fontface="bold" ) + 
  scale_fill_manual(values=newcolours, name = "Legend") +
  scale_color_manual(values=newcolours, name = "Legend") +
  theme_classic() + xlab("Subject") + 
  theme(axis.text.y = element_text(face = "italic")) + ylab("") + 
  theme(legend.position="bottom") +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  scale_y_discrete(limits = rev)
SFig4
ggsave(SFig4, file="~/Desktop/NATURE/ABX_SAMPLES/Manuscript/Submission2/figures/SupplementaryFigure4.png", height = 8, width = 12)
```
